src/main/java/com/flansmod/common/guns/AAGunType.java

```diff
@@ -47,138 +50,113 @@ public class AAGunType extends InfoType
 
 	public static List<AAGunType> infoTypes = new ArrayList<AAGunType>();
 
-	public AAGunType(TypeFile file)
-	{
+	public AAGunType(TypeFile file) {
 		super(file);
 		infoTypes.add(this);
 	}
 
 	@Override
-	protected void preRead(TypeFile file) 
-	{
-	}
+	protected void preRead(TypeFile file) { }
 
 	@Override
-	protected void postRead(TypeFile file) 
-	{		
-	}
+	protected void postRead(TypeFile file) { }
 	
 	@Override
-	protected void read(String[] split, TypeFile file)
-	{
-		super.read(split, file);
-		try
-		{
-			if (FMLCommonHandler.instance().getSide().isClient() && split[0].equals("Model"))
-			{
-				model = FlansMod.proxy.loadModel(split[1], shortName, ModelAAGun.class);
-			}
-			else if (split[0].equals("Texture"))
-			{
-				texture = split[1];
-			}
-			else if (split[0].equals("Damage"))
-			{
-				damage = Integer.parseInt(split[1]);
-			}
-			else if (split[0].equals("ReloadTime"))
-			{
-				reloadTime = Integer.parseInt(split[1]);
-			}
-			else if (split[0].equals("Recoil"))
-			{
-				recoil = Integer.parseInt(split[1]);
-			}
-			else if (split[0].equals("Accuracy"))
-			{
-				accuracy = Integer.parseInt(split[1]);
-			}
-			else if (split[0].equals("ShootDelay"))
-			{
-				shootDelay = Integer.parseInt(split[1]);
-			}
-			else if (split[0].equals("ShootSound"))
-			{
-				shootSound = split[1];
-				FlansMod.proxy.loadSound(contentPack, "aaguns", split[1]);
-			}
-			else if (split[0].equals("ReloadSound"))
-			{
-				reloadSound = split[1];
-				FlansMod.proxy.loadSound(contentPack, "aaguns", split[1]);
-			}
-			else if (split[0].equals("FireAlternately"))
-			{
-				fireAlternately = split[1].equals("True");
-			}
-			else if (split[0].equals("NumBarrels"))
-			{
-				numBarrels = Integer.parseInt(split[1]);
-				barrelX = new int[numBarrels];
-				barrelY = new int[numBarrels];
-				barrelZ = new int[numBarrels];
-			}
-			else if(split[0].equals("Barrel"))
-			{
-				int id = Integer.parseInt(split[1]);
-				barrelX[id] = Integer.parseInt(split[2]);
-				barrelY[id] = Integer.parseInt(split[3]);
-				barrelZ[id] = Integer.parseInt(split[4]);
-			}
-			else if (split[0].equals("Health"))
-			{
-				health = Integer.parseInt(split[1]);
-			}
-			else if (split[0].equals("TopViewLimit"))
-			{
-				topViewLimit = Float.parseFloat(split[1]);
-			}
-			else if (split[0].equals("BottomViewLimit"))
-			{
-				bottomViewLimit = Float.parseFloat(split[1]);
-			}
-			else if (split[0].equals("Ammo"))
-			{
-				BulletType type = BulletType.getBullet(split[1]);
-				if (type != null)
-				{
-					ammo.add(type);
+	protected void read(ConfigMap config, TypeFile file) {
+		super.read(config, file);
+		try {
+			model = FlansMod.proxy.loadModel(modelString, shortName, ModelAAGun.class);
```

Comment:
This might fail on the server side. Even though the server side proxy won't crash, referencing `ModelAAGun.class` might be enough on its own.

Reply:
That's a good point, it was previously if'ed out with Side.Client before I think. I'll test it out.